src/blocks/xrandr.rs

```diff
@@ -0,0 +1,191 @@
+            let mi_line_args: Vec<&str> = mi_line.split_whitespace().collect();
+            if let Some(name) = mi_line_args.get(0) {
+                display = name.trim();
+                if let Some(brightness_raw) = b_line.split(':')
+                                                    .collect::<Vec<&str>>().get(1) {
+                    brightness = (f32::from_str(brightness_raw.trim())
+                                        .expect("Unable to parse brightness string to int.") * 100.0)
+                                    .floor() as u32;
+                }
+            }
+            if let Some(res) = mi_line_args.get(2) {
+                if let Some(resolution) = res.split('+')
+                                          .collect::<Vec<&str>>().get(0) {
+                    monitor_metrics.push(Monitor::new(display,
+                                                      brightness,
+                                                      resolution.trim()));
+                }
+            }
+        }
+        if monitor_metrics.is_empty() {
+            return None;
+        } else {
+            return Some(monitor_metrics);
+        }
+    }
+}
+
+impl Block for Xrandr
+{
+    fn update(&mut self) -> Option<Duration> {
+        if let Some(am) = Xrandr::get_active_monitors() {
+            if let Some(mm) = Xrandr::get_monitor_metrics(&am) {
+                self.monitors = mm;
+                if let Some(m) = self.monitors.get(self.current_idx) {
+                    let brightness_str = m.brightness.to_string();
+                    let values = map!("{display}" => m.name.clone(),
+                                      "{brightness}" => brightness_str,
+                                      "{resolution}" => m.resolution.clone());
+
+                    self.text.set_icon("xrandr");
+                    let mut format_str: &str = "";
+                    if self.resolution {
+                        if self.icons {
+                            format_str = "{display} \u{f185} {brightness} \u{f096} {resolution}";
+                        } else {
+                            format_str = "{display}: {brightness} [{resolution}]";
+                        }
+                    } else {
+                        if self.icons {
+                            format_str = "{display} \u{f185}{brightness}";
+                        } else {
+                            format_str = "{display}: {brightness}";
+                        }
+                    }
+
+                    if let Ok(fmt_template) = FormatTemplate::from_string(String::from(format_str)) {
+                        self.text.set_text(fmt_template.render_static_str(&values));
+                    }
+                }
+            }
+        }
+        Some(self.update_interval.clone())
+    }
+    fn view(&self) -> Vec<&I3BarWidget> {
+        vec![&self.text]
+    }
+    fn click(&mut self, _: &I3barEvent) {
```

Comment:
You currently need to use a ButtonWidget instead of a TextWidget, use the block ID as name and match on your blocks ID.... Otherwise you trigger on every Click on any Block. Look at the other blocks for more info. Will be cleaned and refactored in the future.